File: 1-TwoSum/1-TwoSum.py
# Last updated: 3/14/2026, 11:51:27 AM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map= {}
        for i in range(0, len(nums)):
            temp = target - nums[i]
            if temp in map:
                return [i,map.get(temp)]
            else:
                map[nums[i]] = i

File: 1-TwoSum/1-TwoSum.java
// Last updated: 3/14/2026, 11:51:28 AM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i = 0 ; i < nums.length ; i++){
            int temp = target - nums[i];
            if(map.containsKey(temp)){
                return new int[]{i, map.get(temp)};
            }
            else{
                map.put(nums[i], i);
            }
        }

        return null;
    }
}